kale/embed/gripnet.py

```diff
@@ -241,3 +245,294 @@ def forward(
+
+    def __init_module_supervertex__(self, supervertex: SuperVertex, setting: SuperVertexParaSetting):
+        module_list = torch.nn.ModuleList()
+
+        if not supervertex.if_start_supervertex:
+            # add the external modules from all parent supervertices
+            for in_name in supervertex.in_supervertex_list:
+                in_channels = self.supervertex_module_dict[in_name][-1].out_channels
+
+                assert setting.exter_agg_dim is not None
+                out_channels = setting.exter_agg_dim[in_name]
+                module_list.append(GripNetExternalModule(in_channels, out_channels, supervertex.num_node))
+
+        # add the internal module
+        module_list.append(
+            GripNetInternalModule(
+                supervertex.num_node_feat, supervertex.num_edge_type, supervertex.if_start_supervertex, setting
+            )
+        )
+
+        self.supervertex_module_dict[supervertex.name] = module_list
+
+    def forward(self):
+        for supervertex_name in self.supergraph.topological_order:
+            self.__forward_supervertex__(supervertex_name)
+
+        return self.out_embed_dict[self.task_supervertex_name]
+
+    def __forward_supervertex__(self, supervertex_name: str):
+        assert self.supergraph.supervertex_setting_dict is not None
+
+        supervertex = self.supergraph.supervertex_dict[supervertex_name]
+        model = self.supervertex_module_dict[supervertex_name]
+        mode = self.supergraph.supervertex_setting_dict[supervertex_name].mode
+
+        # internal feature layer
+        x = torch.matmul(supervertex.node_feat, model[-1].embedding)
+
+        # external feature aggregation layers
+        if mode == "add":
+            for idx in range(len(supervertex.in_supervertex_list)):
+                parient_name = supervertex.in_supervertex_list[idx]
+                parient_x = self.out_embed_dict[parient_name]
+                superedge = self.supergraph.superedge_dict[(parient_name, supervertex_name)]
+
+                x += model[idx](parient_x, superedge.edge_index, superedge.edge_weight)
+
+        if mode == "cat":
+            tmp = [x]
+            for idx in range(len(supervertex.in_supervertex_list)):
+                parient_name = supervertex.in_supervertex_list[idx]
+                parient_x = self.out_embed_dict[parient_name]
+                superedge = self.supergraph.superedge_dict[(parient_name, supervertex_name)]
+
+                tmp.append(model[idx](parient_x, superedge.edge_index, superedge.edge_weight))
+            x = torch.cat(tmp, dim=1)
+
+        # internal feature aggregation layers
+        if supervertex.num_edge_type > 1:
+            x = model[-1](
+                x, supervertex.edge_index, supervertex.edge_type, supervertex.range_list, supervertex.edge_weight
+            )
+        else:
+            x = model[-1](x, supervertex.edge_index, supervertex.edge_weight)
+
+        self.out_embed_dict[supervertex_name] = x
```

Comment:
try not use x, say what it is, for all occurrences.

Reply:
It would be better to use `x` here, as it is consistent with the `forward()` function of the `*.nn.modules` in PyTorch and PyG.